homeassistant/components/vivotek/config_flow.py

```diff
@@ -0,0 +1,152 @@
+        vol.Required(CONF_PORT, default=80): cv.port,
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Required(CONF_AUTHENTICATION, default=HTTP_BASIC_AUTHENTICATION): vol.In(
+            [HTTP_BASIC_AUTHENTICATION, HTTP_DIGEST_AUTHENTICATION]
+        ),
+        vol.Required(CONF_SSL, default=False): cv.boolean,
+        vol.Required(CONF_VERIFY_SSL, default=True): cv.boolean,
+        vol.Required(CONF_FRAMERATE, default=2): cv.positive_int,
+        vol.Required(CONF_SECURITY_LEVEL, default=DEFAULT_SECURITY_LEVEL): cv.string,
+        vol.Required(
+            CONF_STREAM_PATH,
+            default=DEFAULT_STREAM_SOURCE,
+        ): cv.string,
+    }
+)
+
+
+class VivotekConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Vivotek IP cameras."""
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            try:
+                await self._async_test_config(user_input)
+            except VivotekCameraError:
+                errors["base"] = "cannot_connect"
+            except Exception:
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+            else:
+                return self.async_create_entry(
+                    title=user_input[CONF_NAME],
+                    data=user_input,
+                )
+
+        data_schema = self.add_suggested_values_to_schema(CONF_SCHEMA, user_input or {})
+        return self.async_show_form(
+            step_id="user",
+            data_schema=data_schema,
+            errors=errors,
+            description_placeholders=DESCRIPTION_PLACEHOLDERS,
+        )
+
+    async def async_step_reconfigure(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the reconfiguration step."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            try:
+                await self._async_test_config(user_input)
+            except VivotekCameraError:
+                errors["base"] = "cannot_connect"
+            except Exception:
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+            else:
+                return self.async_update_reload_and_abort(
+                    self._get_reconfigure_entry(),
+                    data_updates=user_input,
+                )
+
+        data = user_input or self._get_reconfigure_entry().data
+        data_schema = self.add_suggested_values_to_schema(CONF_SCHEMA, data or {})
+        return self.async_show_form(
+            step_id="reconfigure",
+            data_schema=data_schema,
+            errors=errors,
+            description_placeholders=DESCRIPTION_PLACEHOLDERS,
+        )
```

Comment:
```suggestion
        return self.async_show_form(
            step_id="reconfigure",
            data_schema=self.add_suggested_values_to_schema(CONF_SCHEMA, data or {}),
            errors=errors,
            description_placeholders=DESCRIPTION_PLACEHOLDERS,
        )
```
See if you can also easily and readably merge in the `data`. The essence is that the single-use variables can be put in directly, instead of assigning them. :)